client/src/components/transformation/ReverseGeocodeTransformation.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Immutable from 'immutable';
+import TransformationHeader from './TransformationHeader';
+import TargetReverseGeocodeOptions from './reverse-geocode/TargetReverseGeocodeOptions';
+import SourceReverseGeocodeOptions from './reverse-geocode/SourceReverseGeocodeOptions';
+import { ensureDatasetFullyLoaded } from '../../actions/dataset';
+
+import './ReverseGeocodeTransformation.scss';
+
+class ReverseGeocodeTransformation extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      spec: Immutable.fromJS({
+        target: {
+          geopointColumn: null,
+          title: '',
+        },
+        source: {
+          datasetId: null,
+          geopointColumn: null,
+          mergeColumn: null,
+        },
+      }),
+    };
+  }
+
+  componentWillMount() {
+    const { dispatch, datasetId } = this.props;
+    const { spec } = this.state;
+    dispatch(ensureDatasetFullyLoaded(datasetId))
```

Comment:
It's a shame that we are fetching the entire dataset (often several MB) just to get a list of column metadata for a dropdown... but we do exactly the same thing in many other places. 

I also think that if we were being really strict about the redux design pattern, we should move the `dispatch` call to the container and pass a plain callback to this component.

But I think we should address both these things separately - I really think it's time that we seriously looked into loading all column metadata by default, which would resolve both the above problems.

Reply:
Agree on this.